bciers/apps/registration/app/components/operations/OperationRegistrationPage.tsx

```diff
@@ -21,22 +24,21 @@ const OperationRegistrationPage = async ({
   operation: UUID;
   searchParams: FacilitiesSearchParams;
 }) => {
-  // const purpose = operationFormData?.registration_purpose;
-  // hardcoding a value for development; remove value and ts-ignores when feature is implemented
-  const purpose = "New Entrant Operation";
-
-  // Remove steps that aren't applicable to the registration based on purpose
-  let steps = allOperationRegistrationSteps;
-  // @ts-ignore
-  if (purpose !== "New Entrant Operation")
-    steps = steps.filter((e) => e !== "New Entrant Application");
-  // @ts-ignore
-  if (purpose !== "Opted-in Operation")
-    steps = steps.filter((e) => e !== "Opt-in Application");
   let operationData;
-
+  let steps = [...initialOperationRegistrationSteps];
   if (operation && isValidUUID(operation)) {
     operationData = await getOperationV2(operation);
+    const purposes = operationData?.registration_purposes;
+    if (
+      // Note: the purposes have slightly different names than the step names
+      purposes.includes("Opted-in Operation")
+    ) {
+      steps.splice(2, 0, OperationRegistrationSteps.OPT_IN_APPLICATION);
+    }
+    if (purposes.includes("New Entrant Operation"))
+      steps.splice(2, 0, OperationRegistrationSteps.NEW_ENTRANT_APPLICATION);
+  } else {
+    steps = [...initialOperationRegistrationSteps];
```

Comment:
Is this condition necessary?

Reply:
Yes--without it, if someone registers a new entrant/opt-in and then navigates back to register a second operation, the extra steps still appears.